vendor/src/github.com/docker/go-connections/nat/nat.go

```diff
@@ -146,9 +146,31 @@ func ParsePortSpecs(ports []string) (map[Port]struct{}, map[Port][]PortBinding,
 			rawPort = fmt.Sprintf(":%s", rawPort)
 		}
 
-		parts, err := PartParser(portSpecTemplate, rawPort)
-		if err != nil {
-			return nil, nil, err
+		parts := make(map[string]string)
+		if rawPort != "" && rawPort[0] == '[' {
```

Comment:
I saw this PR https://github.com/docker/docker/pull/20842

So wondering if we can make use of the standard net functions here, e.g. [`net. SplitHostPort()`](https://golang.org/pkg/net/#SplitHostPort)